kotlin-utils/src/commonTest/kotlin/org/modelix/kotlin/utils/ContextValueTests.kt

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2023.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.modelix.kotlin.utils
+
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.test.runTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.time.Duration.Companion.milliseconds
+
+class ContextValueTests {
+
+    @Test
+    fun multipleCoroutines() = runTest {
+        val contextValue = ContextValue<String>("a")
+        assertEquals("a", contextValue.getValueOrNull())
+        coroutineScope {
+            launch {
+                for (i in 1..10) {
```

Comment:
Is it really necessary to launch that many instances for verifying the desired behavior?

Reply:
The loop is inside the launch. It doesn't launch new coroutines.

Reply:
But why is it required to loop 10 times? Is there a chance to simplify the test here and to limit it to the bare minimum required to verify the condition being tested here?

Reply:
It tries to find a race-condition between the coroutines. Since a successful test doesn't proof the correctness there is no minimum value. The higher the value the more likely it is that it detects such a race-condition.

Reply:
Too bad [thread-weaver](https://github.com/google/thread-weaver) isn't a thing anymore...

Can you add a comment explaining exactly this? That would make the code a lot more understandable.